homeassistant/components/notify/discord.py

```diff
@@ -0,0 +1,58 @@
+"""Discord platform for notify component."""
+import logging
+import asyncio
+import voluptuous as vol
+import homeassistant.helpers.config_validation as cv
+from homeassistant.components.notify import (
+    PLATFORM_SCHEMA, BaseNotificationService)
+
+_LOGGER = logging.getLogger(__name__)
+
+REQUIREMENTS = ['discord.py']
+
+CONF_TOKEN = 'token'
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_TOKEN): cv.string
+})
+
+
+def get_service(hass, config):
+    """Get the Discord notification service."""
+    try:
+        client_id = config.get(CONF_TOKEN)
+    except:
+        _LOGGER.error("Please specify a token")
+        return None
+    return DiscordNotificationService(hass, client_id)
+
+
+class DiscordNotificationService(BaseNotificationService):
+    """Implement the notification service for Discord."""
+
+    def __init__(self, hass, client_id):
+        """Initialize the service."""
+        self.client_id = client_id
+        self.hass = hass
+
+    async def async_send_message(self, message, target):
+        """Login to Discord and send message."""
+        import discord
+        discord_bot = discord.Client(loop=self.hass.loop)
+
+        """Logs in."""
+        await discord_bot.login(self.client_id)
+
+        """Gets channel ID(s) and sends message."""
+        for channelid in target:
+            channel = discord.Object(id=channelid)
+            await discord_bot.send_message(channel, message)
```

Comment:
We do not support the `async` / `await` syntax. We aim Python 3.4.2+ so use `yield from`